File: Searching Algorithms/Binary Search.cpp
#include <iostream>

using namespace std;

int binarySearch( int A[], int N,int n );

int main()
{
    int N;

    cout << "Enter No of Elements in array : ";
    cin >> N;

    int arr[N];
    int n;

    cout << "Enter elements of array : ";
    for ( int i = 0; i < N; i++ )
    {
        cin >> arr[i];
    }

    cout << "Enter the number to be searched : ";
    cin >> n;

    int pos = binarySearch( arr, N ,n );

    if( pos > 0 )
    {
        cout << "Elements is found at position " << pos << endl;
    }else
    {
        cout << "Element not found" << endl;
    }

    return 0;
}

int binarySearch( int A[], int N, int n )
{
    int ll = 0;
    int ul = N-1;

    int mid = (ll + ul) / 2;

    while ( ll < ul && A[mid] != n )
    {
        if ( A[mid] < n )
        {
            ll = mid + 1;
        }else
        {
            ul = mid - 1;
        }

        mid = (ll + ul) /2;
    }

    if ( A[mid] == n )
    {
        return mid;
    }else
    {
        return -1;
    }
}

File: Searching Algorithms/Python/Binary Search.py
def binary_search( arr, len, num ) :
    ll = 0
    ul = len - 1
    mid = int( ll + ul / 2 )
    while( ll < ul and mid < len ) :
        if arr[mid] > num :
            ul = mid - 1
        elif arr[mid] < num :
            ll = mid + 1
        else:
            break;
        mid = int( ul + ll / 2 )

    if( mid < len ) :
        if( arr[mid] == num ) :
            return mid
    else :
        return -1

len = int( input( "Enter the length of array : " ) )

arr = []

print( "Enter Values in Ascending Order" )
for i in range( len ) :
    arr.append( int( input( "Enter value in array : " ) ) )

num = int( input( "Enter the number to find : " ) )

loc = binary_search( arr, len, num )

if( loc != -1 ) :
    print( "Number is found at index {}".format( loc ) )
else :
    print( "Number is not found in given array" )